libs/features/mas/web-components/src/hydrate.js

```diff
@@ -12,29 +12,33 @@ export async function hydrate(fragmentData, merchCard) {
     if (!variant) return;
     fragment.model = fragment.model;
 
-    merchCard.variantLayout?.refs?.forEach((ref) => ref.remove());
+    // remove all previous slotted content except the default slot
+    merchCard.querySelectorAll('[slot]').forEach((el) => {
+        el.remove();
+    });
     merchCard.variant = variant;
     await merchCard.updateComplete;
-    merchCard.variantLayout.refs ??= [];
+
     const { aemFragmentMapping } = merchCard.variantLayout;
 
     if (!aemFragmentMapping) return;
 
     const appendFn = (el) => {
-        merchCard.variantLayout.refs.push(el);
```

Comment:
actually, what was merchCard.variantLayout.refs? what was `merchCard.variantLayout.refs.push(el); `needed for?

Reply:
with `merchCard.variantLayout.refs` I initially tried to track elements that were added, but it seemed a bit unnecessary and clearing all named slots is easier and safer. (especially when you change card variant in MAS Studio)

Reply:
got it